sdk/documentintelligence/azure-ai-documentintelligence/azure/ai/documentintelligence/_operations/_patch.py

```diff
@@ -266,7 +321,361 @@ def get_long_running_output(pipeline_response):
+        kwargs.pop("error_map", None)
+
+        def get_long_running_output(pipeline_response):
+            response_headers = {}
+            response = pipeline_response.http_response
+            response_headers["Retry-After"] = self._deserialize("int", response.headers.get("Retry-After"))
+            response_headers["Operation-Location"] = self._deserialize(
+                "str", response.headers.get("Operation-Location")
+            )
+
+            deserialized = _deserialize(_models.AnalyzeResult, response.json().get("analyzeResult"))
+            if cls:
+                return cls(pipeline_response, deserialized, response_headers)  # type: ignore
+            return deserialized
+
+        path_format_arguments = {
+            "endpoint": self._serialize.url("self._config.endpoint", self._config.endpoint, "str", skip_quote=True),
+        }
+
+        if polling is True:
+            polling_method = AnalyzeDocumentLROPollingMethod(
+                lro_delay, path_format_arguments=path_format_arguments, **kwargs
+            )
+        elif polling is False:
+            polling_method = cast(AnalyzeDocumentLROPollingMethod, NoPolling())
+        else:
+            polling_method = polling
+        if cont_token:
+            return cast(
+                AnalyzeDocumentLROPoller, LROPoller[_models.AnalyzeResult].from_continuation_token(
+                    polling_method=polling_method,
+                    continuation_token=cont_token,
+                    client=self._client,
+                    deserialization_callback=get_long_running_output,
+                )
+            )
+        return AnalyzeDocumentLROPoller[_models.AnalyzeResult](
+            self._client, raw_result, get_long_running_output, polling_method  # type: ignore
+        )
+
+    @distributed_trace
+    def begin_analyze_batch_documents(
+        self,
+        model_id: str,
+        analyze_batch_request: Optional[Union[_models.AnalyzeBatchDocumentsRequest, JSON, IO[bytes]]] = None,
+        *,
+        pages: Optional[str] = None,
+        locale: Optional[str] = None,
+        string_index_type: Optional[Union[str, _models.StringIndexType]] = None,
+        features: Optional[List[Union[str, _models.DocumentAnalysisFeature]]] = None,
+        query_fields: Optional[List[str]] = None,
+        output_content_format: Optional[Union[str, _models.ContentFormat]] = None,
+        output: Optional[List[Union[str, _models.AnalyzeOutputOption]]] = None,
+        **kwargs: Any,
+    ) -> LROPoller[_models.AnalyzeBatchResult]:
+        _headers = case_insensitive_dict(kwargs.pop("headers", {}) or {})
+        _params = kwargs.pop("params", {}) or {}
+
+        content_type: Optional[str] = kwargs.pop("content_type", _headers.pop("content-type", None))
+        cls: ClsType[_models.AnalyzeBatchResult] = kwargs.pop("cls", None)
+        polling: Union[bool, PollingMethod] = kwargs.pop("polling", True)
```

Comment:
if it is just the polling method that needs to be customized for this operation, you can just pass it using the `polling` keyword argument and save yourself from copying all the generated code.

super().begin_analyze_batch_documents(..., polling=AnalyzeBatchDocumentsLROPollingMethod(...))

Reply:
Updated